### PR TITLE
add matching doc to errorkind

### PR DESCRIPTION
**Pull request by Dylan-DPC**
Rework of #90706

### CODE REVIEW

File: library/std/src/io/error.rs [hunk 1]
```diff
@@ -141,6 +141,16 @@ struct Custom {
...
 ///
 /// [`io::Error`]: Error
+///
+/// # Handling errors and matching on `ErrorKind`
+///
+/// In application code, use `match` for the `ErrorKind` values you are expecting; use `_` to match
+/// "all other errors".
+///
+/// In comprehensive and thorough tests that want to verify that a test doesn't return any known incorrect error kind,
+/// you may want to cut-and-paste the current list of errors from here into your test code. This seems counterintuitive,
```
Comment (joshtriplett) [inline library/std/src/io/error.rs:151]:
```suggestion
/// you may want to cut-and-paste the current full list of errors from here into your test code, and then match `_` as the correct case. This seems counterintuitive,
```

(modulo word wrapping)

**Comment by joshtriplett** [comment]
r=me with something like that change incorporated.

**Comment by joshtriplett** [comment]
(needs a word-wrap now, but LGTM otherwise)

**Comment by Dylan-DPC** [comment]
ah i just committed the suggestion from github ui :D

**Pull request merged**